urnai/base/persistence.py

```diff
@@ -0,0 +1,83 @@
+import os
+from abc import ABC, abstractmethod
+from multiprocessing import Process
+
+
+class Persistence(ABC):
+    """
+    This interface represents the concept of a class that can be saved to disk.
+    The heir class should define a constant or attribute as a default filename
+	to save on disk.
+    """
+
+    def __init__(self, threaded_saving=False):
+        self.threaded_saving = threaded_saving
+        self.attr_block_list = []
+
+    def _get_default_save_stamp(self):
+        """
+        This method returns the default
+        file name that should be used while
+        persisting the object.
+        """
+        return self.__class__.__name__ + '_'
+
+    def get_full_persistance_path(self, persist_path):
+        """This method returns the default persistance path."""
+        return persist_path + os.path.sep + self._get_default_save_stamp()
+
+    def save(self, savepath):
+        if self.threaded_saving:
+            self._threaded_save(savepath)
+        else:
+            self._save(savepath)
+
+    def _threaded_save(self, savepath):
+        """
+        This method saves pickleable
+        stuff in a separate thread
+        """
+        new_process = Process(target=self._save, args=(savepath,))
+        new_process.start()
+
+    @abstractmethod
+    def _save(self, savepath):
+        """
+        This method handles logic related
+        to the child class
+        """
+        ...
+
+    def load(self, loadpath):
+        self._load(loadpath)
+
+    @abstractmethod
+    def _load(self, savepath):
+        """
+        This method handles logic related
+        to the child class
+        """
+        ...
```

Comment:
Instead, you can simply set `load` to `abstractmethod`.

```suggestion
    @abstractmethod
    def load(self, loadpath):
        """
        This method handles logic related
        to the child class
        """
        ...
```